index.html

```diff
@@ -7,54 +7,71 @@
+
+        <div class="options">
+          <div class="numb">
+            <label>log each</label>
+            <input title="'log each' defines the interval at which blocks are printed. A value of 2 means that we print one block out of two. A value of 1 means all the blocks are printed." type="number" id="log-each-input" value="1">
+          </div>
+          <a style="display: none" id="download-link">Download results</a>
+          <div class="numb">
+            <label>print details</label>
+            <input title="Prints the details of each block. Is also used to display more details for the status." id="details-input" type="checkbox">
+          </div>
+          <div class="numb">
+            <label>print roster</label>
+            <input title="If checked, the roster is printed on each block, as well as when the status is fetched." id="roster-input" type="checkbox">
+          </div>
+          <div class="numb">
+            <label>repeat cmd</label>
+            <input title="The number of times the 'Load', '◀︎', or '▶︎' is repeated. This features is mostly interesting from the code side." type="number" id="repeat-cmd-input" value="0">
+          </div>
+          <button title="Fetch the latest block starting from the latest loaded block, or the one specified in the 'First block to fetch' input if none have been loaded yet. If follows the highest forward link on each block, until reaching a block with no forward link." id="get-latest">Get latest</button>
+          <button title="Fetch the roster from the latest block and display the status of each node. Takes into account the 'print details' and 'print roster' options. If no blocks have been loaded, it gets the latest block starting from the one in the 'First block to fetch' input. Otherwise, it takes the latest block loaded." id="get-status">Get status</button>
         </div>
-        <button title="Fetch the latest block starting from the latest loaded block, or the one specified in the 'First block to fetch' input if none have been loaded yet. If follows the highest forward link on each block, until reaching a block with no forward link." id="get-latest">Get latest</button>
-        <button title="Fetch the roster from the latest block and display the status of each node. Takes into account the 'print details' and 'print roster' options. If no blocks have been loaded, it gets the latest block starting from the one in the 'First block to fetch' input. Otherwise, it takes the latest block loaded." id="get-status">Get status</button>
       </div>
 
+      <div class="sep"></div>
+
+      <div class="bypros">
+        <textarea title="Here you can submit an SQL query" type="text" id="sql-input" cols="33" rows="9">select
+  encode(hash, 'hex')
+    as hash,
+  block_id
+  from cothority.block 
+  limit 2;</textarea>
```

Comment:
Is there a limit to the number of blocks we can fetch? I tried a `limit 10000` and got 109 blocks.

Also how do the indexes map to block ids? "Get latest" returned

> latest block found: 2ad9c38c76a22403242c0a04d950ef913432a4b9f90c0144268abbbc799f9e47 with index 173243

The SQL query reported the same last block with a block id of 109.

Edit: Is this because I didn't catch up?

Reply:
Yes, I didn't do a full catch up yet. For the index, the database doesn't store yet the block index. The block_id is nothing more than the (useless) primary key of the entry in the DB.